File: app/src/main/java/hedera/hgc/hgcwallet/app_intent/LinkAccountRequestParams.java
package hedera.hgc.hgcwallet.app_intent;

import android.net.Uri;

import org.json.JSONObject;

public class LinkAccountRequestParams {
    public Uri callback, redirect;

    public LinkAccountRequestParams (Uri callback,  Uri redirect) {
        this.callback = callback;
        this.redirect = redirect;
    }
    public static LinkAccountRequestParams from(JSONObject jsonObject) {
        if (jsonObject != null) {
            String action = jsonObject.optString("action","");
            if (action.equals("requestPublicKey")) {
                String callback = jsonObject.optString("callback","");
                String redirect = jsonObject.optString("redirect","");
                Uri callbackUri = Uri.parse(callback);
                Uri redirectUri = Uri.parse(redirect);
                if (callback != null) {
                    return new LinkAccountRequestParams(callbackUri, redirectUri);
                }
            }

        }
        return null;
    }
}

File: app/src/main/java/hedera/hgc/hgcwallet/app_intent/LinkAccountRequestParams.kt
package hedera.hgc.hgcwallet.app_intent

import android.net.Uri

import org.json.JSONObject

class LinkAccountRequestParams(val callback: Uri, val redirect: Uri?) {
    companion object {
        fun from(jsonObject: JSONObject?): LinkAccountRequestParams? {
            if (jsonObject != null) {
                val action = jsonObject.optString("action", "")
                if (action == "requestPublicKey") {
                    val callback = jsonObject.optString("callback", "")
                    val redirect = jsonObject.optString("redirect", "")
                    val callbackUri = Uri.parse(callback)
                    val redirectUri = Uri.parse(redirect)
                    if (callback != null) {
                        return LinkAccountRequestParams(callbackUri, redirectUri)
                    }
                }

            }
            return null
        }
    }
}










